ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -175,7 +177,8 @@ data ShakeExtras = ShakeExtras
     ,publishedDiagnostics :: Var (HMap.HashMap NormalizedUri [Diagnostic])
     -- ^ This represents the set of diagnostics that we have published.
     -- Due to debouncing not every change might get published.
-    ,positionMapping :: Var (HMap.HashMap NormalizedUri (Map TextDocumentVersion (PositionDelta, PositionMapping)))
+    ,positionMapping :: Var (HMap.HashMap NormalizedUri (Map TextDocumentVersion (PositionDelta, PositionMapping, WeakToken)))
+    ,liveTokens :: Var (HMap.HashMap NormalizedUri (Token, TextDocumentVersion, WeakToken))
```

Comment:
I don't see the `liveTokens` map being pruned anywhere. How does this scheme work then?